### PR TITLE
Issue177：Upgrade go.mod & LICENSE

### LINKED ISSUE #177: some issues needed to fix for v1.5.0
**What would you like to be added**:

1. appendix in LICENSE

2. update the dependency libs with them latest version in go.mod:
- github.com/dubbogo/gost v1.8.0
- github.com/pkg/errors v0.9.1
- github.com/stretchr/testify v1.5.1

3. add comment for public struct/functions in
- java_collection.go
- request.go
- serialize.go

### PR DESCRIPTION
**Pull request by micln**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #177 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by micln** [comment]
@AlexStocks Please review go.sum and License

**Comment by AlexStocks** [comment]
pls check the travis error.

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,9 @@ matrix:
       jdk: openjdk8
 
 go:
-  - "1.12"
+  - "1.12.x"
```
Comment (AlexStocks) [inline .travis.yml:9]: just one go version is enough.

Comment (micln) [reply .travis.yml:9]: the dependend libs may need diffirent version. multi version testing may help us find the errors caused by lib-version

**Pull request merged**